ethereum/stratum/src/main/java/org/hyperledger/besu/ethereum/stratum/StratumConnection.java

```diff
@@ -32,103 +28,52 @@
 final class StratumConnection {
   private static final Logger LOG = LoggerFactory.getLogger(StratumConnection.class);
 
-  private String incompleteMessage = "";
-  private boolean httpDetected = false;
-
   private final StratumProtocol[] protocols;
   private final Runnable closeHandle;
-  private final Consumer<String> sender;
-
+  private final Consumer<String> notificationSender;
   private StratumProtocol protocol;
 
-  private static final Pattern contentLengthPattern =
-      Pattern.compile("\r\nContent-Length: (\\d+)\r\n");
-
   StratumConnection(
       final StratumProtocol[] protocols,
       final Runnable closeHandle,
-      final Consumer<String> sender) {
+      final Consumer<String> notificationSender) {
     this.protocols = protocols;
     this.closeHandle = closeHandle;
-    this.sender = sender;
+    this.notificationSender = notificationSender;
   }
 
-  void handleBuffer(final Buffer buffer) {
-    LOG.trace("Buffer received {}", buffer);
-    String messagesString;
-    try {
-      messagesString = buffer.toString(StandardCharsets.UTF_8);
-    } catch (IllegalArgumentException e) {
-      LOG.debug("Invalid message with non UTF-8 characters: " + e.getMessage(), e);
-      closeHandle.run();
-      return;
-    }
-
-    if (httpDetected) {
-      httpDetected = false;
-      messagesString = incompleteMessage + messagesString;
-    }
-    Matcher match = contentLengthPattern.matcher(messagesString);
-    if (match.find()) {
-      try {
-        int contentLength = Integer.parseInt(match.group(1));
-        String body = messagesString.substring(messagesString.indexOf("\r\n\r\n") + 4);
-        if (body.length() < contentLength) {
-          incompleteMessage = messagesString;
-          httpDetected = true;
-          return;
-        }
-      } catch (NumberFormatException e) {
-        close();
-        return;
-      }
-
-      LOG.trace("Dispatching HTTP message {}", messagesString);
-      handleMessage(messagesString);
+  void handleBuffer(final Buffer message, final Consumer<String> sender) {
+    final Buffer trimmedMessage;
+    if (message.getByte(message.length() - 1)
+        == System.lineSeparator().getBytes(StandardCharsets.UTF_8)[0]) {
```

Comment:
That's weird. Is `\r` a valid separator?

Reply:
No, you're correct. I missed the case of `line.separator` being `\r\n` for Windows systems. I'll fix it.

Reply:
I fixed this and moved this code to the line based pipeline b/c this trim up doesn't make sense for HTTP